### PR TITLE
Cleanup Manager register reference with AttrDict implementation

### PR DESCRIPTION
**Pull request by jjhong922**
Makes data_registry, state_registry, and summary_stats properties return attrdicts which allow users to reference dictionary keys using attribute notation. This cleans up verbose references to field constants without importing the original field class.

### CODE REVIEW

File: scvi/utils/_attrdict.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+from copy import deepcopy
+
+
+class attrdict(dict):
```
Comment (adamgayoso) [inline scvi/utils/_attrdict.py:4]: we will probably have to ignore this on codacy

File: scvi/utils/_attrdict.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    """
+    A dictionary that allows for attribute notation (e.g. d.element).
+
+    Based off of https://stackoverflow.com/questions/38034377/object-like-attribute-access-for-nested-dictionary.
```
Comment (adamgayoso) [inline scvi/utils/_attrdict.py:8]: Put in Notes section like

```
Notes
-----
...
```

File: scvi/utils/__init__.py [hunk 3]
```diff
@@ -1,4 +1,5 @@
+from ._attrdict import attrdict
 from ._docstrings import setup_anndata_dsp
 from ._track import track
 
-__all__ = ["track", "setup_anndata_dsp"]
+__all__ = ["track", "setup_anndata_dsp", "attrdict"]
```
Comment (adamgayoso) [inline scvi/utils/__init__.py:5]: we need to add `attrdict` to the docs page

File: scvi/utils/_attrdict.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (adamgayoso) [reply scvi/utils/_attrdict.py:4]: also should we overwrite the repr? To better communicate there that it's an attrdict and not a dict?

[the remark this answers is not in the document]
Comment (adamgayoso) [reply scvi/utils/_attrdict.py:4]: would help devs see what they can do with it

**Review by adamgayoso — CHANGES_REQUESTED** [review]

File: scvi/utils/_attrdict.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jjhong922) [reply scvi/utils/_attrdict.py:4]: the check passed on codacy, did you ignore something? And yes I'll overwrite the repr can just append something to the front of it

**Pull request merged**